test/google_search_data_viewer/keyword/search_results_test.exs

```diff
@@ -54,4 +54,108 @@ defmodule GoogleSearchDataViewer.Keyword.SearchResultsTest do
              }
     end
   end
+
+  describe "group_by_adword_types/1" do
+    test "given SearchResultUrls with two URLs that are not adwords, returns a Map with two non_adwords with the key non_adwords" do
+      keyword_upload = insert(:keyword_upload)
+
+      search_results =
+        insert_list(2, :search_result_url,
+          is_adword: false,
+          is_top_adword: false,
+          keyword_upload: keyword_upload
+        )
+
+      result = SearchResults.group_by_adword_types(search_results)
+
+      assert Map.has_key?(result, :adwords) == false
+      assert Map.has_key?(result, :top_adwords) == false
+      assert Enum.count(result.non_adwords) == 2
+    end
+
+    test "given SearchResultUrls with one URL that is an adword, returns a Map with one adword with the key adwords" do
+      keyword_upload = insert(:keyword_upload)
+
+      search_results =
+        insert_list(1, :search_result_url,
+          is_adword: true,
+          is_top_adword: false,
+          keyword_upload: keyword_upload
+        )
+
+      result = SearchResults.group_by_adword_types(search_results)
+
+      assert Map.has_key?(result, :non_adwords) == false
+      assert Map.has_key?(result, :top_adwords) == false
+      assert Enum.count(result.adwords) == 1
+    end
+
+    test "given SearchResultUrls with two URLs that are top_adwords, returns a Map with two top_adwords with the key top_adwords" do
+      keyword_upload = insert(:keyword_upload)
+
+      search_results =
+        insert_list(2, :search_result_url,
+          is_adword: true,
+          is_top_adword: true,
+          keyword_upload: keyword_upload
+        )
+
+      result = SearchResults.group_by_adword_types(search_results)
+
+      assert Map.has_key?(result, :adwords) == false
+      assert Map.has_key?(result, :non_adwords) == false
+      assert Enum.count(result.top_adwords) == 2
+    end
+
+    test "given SearchResultUrls with two URLs that are a top_adword and adword, returns a Map with both keys" do
+      keyword_upload = insert(:keyword_upload)
+
+      search_results = [
+        insert(:search_result_url,
+          is_adword: true,
+          is_top_adword: true,
+          keyword_upload: keyword_upload
+        ),
+        insert(:search_result_url,
+          is_adword: true,
+          is_top_adword: false,
+          keyword_upload: keyword_upload
+        )
```

Comment:
In this case we can create multiple factories `insert(:url_top_ads...`, `insert(:url_ads...`

Reply:
Could you elaborate a little bit on that

Reply:
Included here 0d7cbe4ff050f66a058b817188736fdaa49416cf